### PR TITLE
Course creation flow

### PR DESCRIPTION
**Pull request by OlgaLopushanska**

### CODE REVIEW

File: src/main/java/com/itacademy/cms/controller/CategoryController.java [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package com.itacademy.cms.controller;
 
 import com.itacademy.cms.exeption.CategoryNotFoundException;
-import com.itacademy.cms.mapper.EntityMapper;
+import com.itacademy.cms.mapper.MapStructMapper;
```
Comment (IvanZabor) [inline src/main/java/com/itacademy/cms/controller/CategoryController.java:4]: We seem to have agreed to name Mapper as EntityMapper

Comment (OlgaLopushanska) [reply src/main/java/com/itacademy/cms/controller/CategoryController.java:4]: it look like i have miss it

File: src/main/java/com/itacademy/cms/exeption/CategoryNotFoundException.java [hunk 2]
```diff
@@ -1,6 +1,6 @@
 package com.itacademy.cms.exeption;
 
-public class CategoryNotFoundException extends RuntimeException {
+public class CategoryNotFoundException extends Exception {
```
Comment (IvanZabor) [inline src/main/java/com/itacademy/cms/exeption/CategoryNotFoundException.java:3]: Please change Exception to RuntimeException

Comment (OlgaLopushanska) [reply src/main/java/com/itacademy/cms/exeption/CategoryNotFoundException.java:3]: i thing that all exceptions must be the same type, so i did all of them the same type

File: src/main/java/com/itacademy/cms/model/Category.java [hunk 3]
```diff
@@ -20,9 +18,7 @@
 @Setter
 public class Category extends BaseEntity {
 
-  @OneToMany(cascade = CascadeType.ALL)//mappedBy = "category", fetch = FetchType.LAZY)
+  @OneToMany(mappedBy = "category", fetch = FetchType.LAZY)
```
Comment (IvanZabor) [inline src/main/java/com/itacademy/cms/model/Category.java:21]: If we use mappedBy with fetch, at the endpoint category will recursively call course and RunnOutOfMemmory

Comment (OlgaLopushanska) [reply src/main/java/com/itacademy/cms/model/Category.java:21]: Ivan you also can do, i am not in charge of Category endpoint I am just trying to put all together and make it work

File: src/main/java/com/itacademy/cms/controller/CategoryController.java (hunk 1 above)
Comment (IvanZabor) [reply src/main/java/com/itacademy/cms/controller/CategoryController.java:4]: No problem, it can be MapStructMapper either

**Review by IvanZabor — APPROVED** [review]

**Pull request merged**